internal/provider/resource_pipeline.go

```diff
@@ -0,0 +1,366 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/humanitec/humanitec-go-autogen"
+	"github.com/humanitec/humanitec-go-autogen/client"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces
+var _ resource.Resource = &ResourcePipeline{}
+var _ resource.ResourceWithImportState = &ResourcePipeline{}
+
+func NewResourcePipeline() resource.Resource {
+	return &ResourcePipeline{}
+}
+
+// ResourceRule defines the resource implementation.
+type ResourcePipeline struct {
+	client *humanitec.Client
+	orgID  string
+}
+
+func (r *ResourcePipeline) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_pipeline"
+}
+
+func (r *ResourcePipeline) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "An Automation Rule defining how and when artefacts in an environment should be updated.",
+
+		Attributes: map[string]schema.Attribute{
+			"app_id": schema.StringAttribute{
+				MarkdownDescription: "The ID of the Application that the Rule should belong to.",
+				Required:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"definition": schema.StringAttribute{
+				MarkdownDescription: "The YAML definition of the pipeline",
+				Required:            true,
+			},
+			"id": schema.StringAttribute{
+				MarkdownDescription: "The Name of the pipeline",
+				Computed:            true,
+			},
+			"created_at": schema.StringAttribute{
+				MarkdownDescription: "The Name of the pipeline",
```

Comment:
These fields are still using "name" - I think you should update them.

Reply:
Done :)